src/system/SystemPacketBuffer.cpp

```diff
@@ -236,6 +241,12 @@ void PacketBuffer::SetDataLength(uint16_t aNewLen, PacketBuffer * aChainHead)
 
 uint16_t PacketBuffer::MaxDataLength() const
 {
+#if CHIP_SYSTEM_CONFIG_USE_LWIP
+    if (!(PBUF_STRUCT_DATA_CONTIGUOUS(this)))
+    {
+        return static_cast<uint16_t>(DataLength());
```

Comment:
Why is that cast needed?  `DataLength` returns `uint16_t`.

Reply:
Removed the cast